File: src/main/java/br/com/teste/java/testebackend/domain/Linha.java
package br.com.teste.java.testebackend.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
@Entity
public class Linha implements Serializable {

    private static final long serialVersionUID = -3656431259068389491L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="ID_LINHA", unique = true, nullable = false)
    private Long id;


    private String name;
    private double latitude;
    private double longitude;

    @ManyToMany(cascade = CascadeType.ALL)
    @JoinTable(
            name = "LINHA_PARADA",
            joinColumns = @JoinColumn(name = "linha_id"),
            inverseJoinColumns = @JoinColumn(name = "parada_id")
    )
    private List<Parada> paradas = new ArrayList();

}

File: teste-back-end/src/entity/Linha.ts
import {
  Entity, PrimaryGeneratedColumn, Column, ManyToMany, JoinTable, OneToMany,
} from 'typeorm';
import Parada from './Parada';
import Veiculo from './Veiculo';

@Entity()
export default class Linha {
  constructor(name: string) {
    this.name = name;
  }

  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  name: string;

  @ManyToMany(() => Parada)
  @JoinTable()
  paradas: Parada[];

  @OneToMany(() => Veiculo, (veiculo) => veiculo.linha)
  veiculos: Veiculo[]
}
